lib/src/components/NxTextInput/NxTextInput.scss

```diff
@@ -4,19 +4,32 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-@import '../../scss-shared/nx-variables';
 @import '../../scss-shared/_nx-form-validation.scss';
 
-.nx-text-input:not(:disabled) {
-  &.nx-text-input--valid {
-    @include valid();
-  }
-
-  &.nx-text-input--invalid {
+.nx-text-input {
+  &.invalid {
     @include invalid();
   }
 
-  &.nx-text-input--pristine {
+  &.pristine, &.pristine.valid {
     @include pristine();
   }
+
+  &.valid {
+    @include valid();
+  }
+}
+
+.nx-text-input__invalid-message {
+  color: #cd0028;
+  font-size: $nx-font-size-xs;
+  margin-top: $nx-spacing-xs;
+}
+
+.nx-icon--invalid {
+  color: #cd0028;
+}
+
+.nx-icon--valid {
```

Comment:
These two should either be in the base styles or should be scoped to this component (e.g. `.nx-text-input .nx-icon--valid`.  The way it is now, the styles of unrelated `.nx-icon--valid` instances could change based on whether or not there is an `NxTextInput` present in the app

Reply:
> These two should either be in the base styles or should be scoped to this component (e.g. `.nx-text-input .nx-icon--valid`. The way it is now, the styles of unrelated `.nx-icon--valid` instances could change based on whether or not there is an `NxTextInput` present in the app

https://github.com/sonatype/sonatype-react-shared-components/pull/57/commits/7b2daa2af25f66c685960a1ae8f68aa6a63099d3